File: Cocoa Programming/Cocoa Pr L69 ObjC/Cocoa Pr L69 ObjC/CodeWindowController.m
//
//  CodeWindowController.m
//  Cocoa Pr L69 ObjC
//
//  Created by Michele Galvagno on 02/09/22.
//

#import "CodeWindowController.h"
#import "Window.h"
#import "ViewController.h"

@interface CodeWindowController ()

@end

@implementation CodeWindowController
//- (instancetype)initWithWindowNibName:(NSNibName)windowNibName {
//    windowNibName = @"";
//    NSLog(@"%@", windowNibName);
//
//    return self;
//}

- (instancetype)init
{
    self = [super initWithWindowNibName:@"" owner:self];
    
    return self;
}

- (void)loadWindow {
    self.window = [[Window alloc] initWithContentRect:NSMakeRect(100, 100, 400, 300) styleMask:NSWindowStyleMaskBorderless backing:NSBackingStoreBuffered defer:YES];
}

- (void)windowDidLoad {
    [super windowDidLoad];
    
    [self setContentViewController:[[ViewController alloc] init]];
}

@end

File: Cocoa Programming/Cocoa Pr L69 NSWindowController/Cocoa Pr L69 NSWindowController/CodeWindowController.swift
//
//  CodeWindowController.swift
//  Cocoa Pr L69 NSWindowController
//
//  Created by Michele Galvagno on 28/08/22.
//

import Cocoa

class CodeWindowController: NSWindowController {
    convenience init() {
        self.init(windowNibName: "")
    }
    
    override func loadWindow() {
        self.window = Window(contentRect: NSMakeRect(100, 100, 400, 300), styleMask: [], backing: .buffered, defer: true)
    }

    override func windowDidLoad() {
        super.windowDidLoad()
        
        contentViewController = ViewController()
    }
    
}
